Library/Homebrew/formula.rb

```diff
@@ -2886,7 +2886,7 @@ def run_test(keep_tmp: false)
 
   sig { returns(T::Boolean) }
   def test_defined?
-    false
+    method(:test).owner != Formula
```

Comment:
This implementation matches [post_install_defined?](https://github.com/Homebrew/brew/blob/3b1d59d8c4fbf8c0ba0c9752007ff27bec3bc8b3/Library/Homebrew/formula.rb#L1395), and no longer requires us to define a `method_added` hook.